L1Trigger/L1TGlobal/plugins/GtExtCondLegacyToStage2.cc

```diff
@@ -0,0 +1,224 @@
+
+  private:
+    virtual void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+
+    // ----------member data ---------------------------
+    //unsigned long long m_paramsCacheId; // Cache-ID from current parameters, to check if needs to be updated.
+    //boost::shared_ptr<const CaloParams> m_dbpars; // Database parameters for the trigger, to be updated as needed.
+    //boost::shared_ptr<const FirmwareVersion> m_fwv;
+    //boost::shared_ptr<FirmwareVersion> m_fwv; //not const during testing.
+
+    // BX parameters
+    int bxFirst_;
+    int bxLast_;
+
+    // Readout Record token
+    edm::EDGetTokenT<L1GlobalTriggerReadoutRecord> gtReadoutRecordToken;
+
+  };
+
+  //
+  // constructors and destructor
+  //
+  GtExtCondLegacyToStage2::GtExtCondLegacyToStage2(const ParameterSet& iConfig) :
+    bxFirst_ (iConfig.getParameter<int>("bxFirst")),
+    bxLast_ (iConfig.getParameter<int>("bxLast")),
+    gtReadoutRecordToken (consumes <L1GlobalTriggerReadoutRecord> (iConfig.getParameter<edm::InputTag>("LegacyGtReadoutRecord")))
+  {
+    // register what you produce
+    produces<GlobalExtBlkBxCollection>();
+
+  }
+
+
+  GtExtCondLegacyToStage2::~GtExtCondLegacyToStage2()
+  {
+  }
+
+
+
+  //
+  // member functions
+  //
+
+  // ------------ method called to produce the data ------------
+  void
+  GtExtCondLegacyToStage2::produce(edm::StreamID, Event& iEvent, const EventSetup& iSetup) const
+  {
+
+    LogDebug("GtExtCondLegacyToStage2") << "GtExtCondLegacyToStage2::produce function called...\n";
+
+    edm::Handle<L1GlobalTriggerReadoutRecord> gtReadoutRecord;
+    iEvent.getByToken(gtReadoutRecordToken, gtReadoutRecord);
+
+
+    // Setup vectors
+    GlobalExtBlk extCond_bx_m2;
+    GlobalExtBlk extCond_bx_m1;
+    GlobalExtBlk extCond_bx_0;
+    GlobalExtBlk extCond_bx_p1;
+    GlobalExtBlk extCond_bx_p2;
+
+    if( gtReadoutRecord.isValid() ){
+      // L1GlobalTriggerReadoutRecord const & l1tResults = * gtReadoutRecord;
+
+      // // select PSB#9 and bunch crossing 0
+      // const L1GtPsbWord & psb = l1tResults.gtPsbWord(0xbb09, 0);
+
+      // // the four 16-bit words psb.bData(1), psb.aData(1), psb.bData(0) and psb.aData(0) yield
+      // // (in this sequence) the 64 technical trigger bits from most significant to least significant bit
+      // uint64_t psbTriggerWord = ( (uint64_t) psb.bData(1) << 48) |
+      // 	((uint64_t) psb.aData(1) << 32) |
+      // 	((uint64_t) psb.bData(0) << 16) |
+      // 	((uint64_t) psb.aData(0));
+
+      // std::cout << "psbTriggerWord = " << psbTriggerWord << std::endl;
+      // //
+
+      for( int ibx = 0; ibx < 5; ibx++ ){
+	int useBx = ibx - 2;
+	if( useBx<bxFirst_ || useBx>bxLast_ ) continue;
+
+	//std::cout << "  BX = " << ibx - 2 << std::endl;
+	
+	// L1 technical
+	const TechnicalTriggerWord& gtTTWord = gtReadoutRecord->technicalTriggerWord(useBx);
+	int tbitNumber = 0;
+	TechnicalTriggerWord::const_iterator GTtbitItr;
+	bool pass_BptxAND = false;
```

Comment:
I would suggest putting these 4 bools into an array - then all the if statements below do not need the if elses